catkit/hub/postgresql.py

```diff
@@ -465,84 +509,118 @@ def transfer(self, filename_sqlite, start_id=1, write_ase=True,
 
             # Publication structures connection
             cur_lite.execute("""SELECT * from publication_system;""")
+            publication_system_values = []
             rows = cur_lite.fetchall()
             for row in rows:
                 Npubstruc += 1
-                values = row[:]
-                key_str, value_str = get_key_value_str(
-                    values, table='publication_system')
-
-                set_schema = 'SET search_path = {0};'.format(self.schema)
-                cur.execute(set_schema)
-                print("[SET SCHEMA] {set_schema}".format(**locals()))
+                values = list(row)
+                value_list = get_value_list(values)
+                publication_system_values += [tuple(value_list)]
 
-                insert_command = """INSERT INTO publication_system ({0})
-                VALUES ({1}) ON CONFLICT DO NOTHING;"""\
-                    .format(key_str, value_str)
+            key_str = get_key_str(table='publication_system')
+            insert_command = """INSERT INTO publication_system ({0})
+            VALUES %s ON CONFLICT DO NOTHING;"""\
+                .format(key_str)
 
-                cur.execute(insert_command)
-                # self.write(values, table='publication_system')
+            execute_values(cur=cur, sql=insert_command,
+                           argslist=publication_system_values, page_size=1000)
             con.commit()
+            self.stdout.write('  Completed transfer of publications\n')
 
         Ncat = 0
         Ncatstruc = 0
 
         if write_reaction:
-            n = db.get_last_id(cur_lite)
-            select_ase = """SELECT * from reaction_system where id={};"""
-            for id_lite in range(start_id, n + 1):
-                row = db.read(id_lite)
-                if len(row) == 0:
-                    continue
-                values = row[0]
+            self.stdout.write('Transfering reactions')
+            cur.execute('SELECT max(id) from reaction;')
+            ID = cur.fetchone()[0] or 0
```

Comment:
`.fetchone()[0]` looks overly optimistic. If `.fetchone()` returns empty list, this would fail and  make testing tricky. Why not something like

    id_fetch = cur.fetchone()
    ID = 0 if (len(id_fetch) == 0) else id_fetch[0]

One extra line for defensive coding ...

Reply:
thanks!